### PR TITLE
py3 compatibility: Ensure map function to return a list

### PR DESCRIPTION
**Pull request by ari3s**
`map()` built-in function returns a list on Python 2, but an iterator
on Python 3. In some context, the problem is solved by adding the `list()`
function. If the first argument to `map()` is a lambda function, then it
is converted to an equivalent list comprehension.

Signed-off-by: Jan Beran <jberan@redhat.com>

### CODE REVIEW

File: src/pyfaf/storage/problem.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def type(self):
 
     @property
     def reports_count(self):
-        return sum(map(lambda x: x.count, self.reports))
+        return sum([x.count for x in self.reports])
```
Comment (xsuchy) [inline src/pyfaf/storage/problem.py]: This is not necessary as sum() can accept iterator. On the other hand list comprehension is probably more readable.

Comment (ari3s) [reply src/pyfaf/storage/problem.py]: Mirek, thank you for the review. My understanding is that you agree with the list comprehension, so I keep the change.

File: src/pyfaf/utils/format.py [hunk 2]
```diff
@@ -27,15 +27,13 @@ def as_table(headers, data, margin=1, separator=' '):
     Return `headers` and `data` lists formatted as table.
     '''
 
-    headers = map(str, headers)
-    data = map(lambda x: map(str, x), data)
+    headers = list(map(str, headers))
+    data = [list(map(str, x)) for x in data]
```
[the remark this answers is not in the document]
Comment (ari3s) [reply src/pyfaf/utils/format.py]: I will fix.

File: src/pyfaf/utils/format.py [hunk 3]
```diff
@@ -27,15 +27,13 @@ def as_table(headers, data, margin=1, separator=' '):
...
 
     widths = reduce(
-        lambda x, y: map(
-            lambda a_b: max(a_b[0], a_b[1]), list(zip(x, y))
-        ),
-        map(lambda x: map(len, x), data) + [map(len, headers)],
-        map(lambda _: 0, headers))
+        lambda x, y: [max(a_b[0], a_b[1]) for a_b in list(zip(x, y))],
+        [list(map(len, x)) for x in data] + [list(map(len, headers))],
```
[the remark this answers is not in the document]
Comment (ari3s) [reply src/pyfaf/utils/format.py]: I will fix.

**Pull request merged**